### ISSUE TITLE
Interactive window names do not match file

### ISSUE THREAD

**Issue by sbland** [microsoft/vscode-jupyter#14052]
Type: <b>Bug</b>

# Behaviour
## Expected vs. Actual
When running a python file cell in an interactive window the window should have the same name as the file. Instead it is named `interactive-1`. When reopening a vscode window with interactive windows open and running a cell in a python file it will run in `interactive-1` even if it previously ran in `interactive-2`. Right clicking the interactive window and renaming doesn't change the tab name.

## Steps to reproduce:

1. Open a python file with runable cells. 
2. Run a cell

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.10
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Venv
-   Value of the `python.languageServer` setting: Pylance

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```

languageServer: "Pylance"

linting
• pylintArgs: "<placeholder>"

```

</p>
</details>

Extension version: 2023.12.0
VS Code version: Code 1.80.2 (2ccd690cbff1569e4a83d7c43d45101f817401dc, 2023-07-27T20:41:52.615Z)
OS version: Linux x64 5.4.0-155-generic snap
Modes:
Remote OS version: Linux x64 3.10.0-1160.92.1.el7.x86_64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz (12 x 3185)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|1, 1, 1|
|Memory (System)|62.75GB (39.20GB free)|
|Process Argv|--no-sandbox --force-user-env --unity-launch --crash-reporter-id ebc65e20-a9f1-4bca-baba-a37d7a051f72|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|

|Item|Value|
|---|---|
|Remote|SSH: arc4.leeds.ac.uk|
|OS|Linux x64 3.10.0-1160.92.1.el7.x86_64|
|CPUs|Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (40 x 1010)|
|Memory (System)|187.37GB (168.75GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
py29gd2263:30792226
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
vsccc:30610678
2e4cg342:30602488
f6dab269:30613381
a9j8j154:30646983
showlangstatbar:30737416
vsctsb:30748421
03d35959:30757346
pythonfmttext:30731395
pythoncmv:30756943
fixshowwlkth:30771522
showindicator:30785052
pythongtdpath:30769146
i26e3531:30792625
gsofa:30797621
pythonnosmt12:30797651
pythonnoceb:30797650
e537b577:30795824
dsvsc013:30795093
dsvsc014cf:30797590

```

</details>

**Comment by davidgilbertson** [comment]
I don't know what's changed since yesterday, but I just noticed that sometimes my tab names have the name of the file!
![image](https://github.com/microsoft/vscode-jupyter/assets/4443482/baafe9c6-59a5-46ce-932f-5b7c4ca5b7b7)

**Comment by amunger** [comment]
This is likely fixed in the latest release, 1.81, can you try there and check if it is fixed?

The tab should have the source file in the title with `"jupyter.interactiveWindow.creationMode": "perFile"`, otherwise, multiple files can contribute to the same Interactive window.

**Comment by sbland** [comment]
Works for me thanks!
